### PR TITLE
[T3A3][T14-A2] Ng Qing Hua

### PR DESCRIPTION
**Pull request by qhng**
Split address to Block, Street, Unit and Postal Code

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -25,7 +30,50 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+    	postalCode = new PostalCode("");
+    }
+    
+    /**
+     * Splits the address into Block, Street, Unit and Postal Code
+     * 
+     * @throws IllegalValueException if given address string is invalid.
+     */
+    private void splitAddress(String address) throws IllegalValueException {
+    	String[] delimitedAddress = address.split(",");
```
Comment (nishantbudhdev) [inline src/seedu/addressbook/data/person/Address.java]: Indentation issue?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -37,19 +85,20 @@ public static boolean isValidAddress(String test) {
...
+        return String.format("%s, %s, %s, %s", 
+        		block.value, street.value, unit.value, postalCode.value);
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.state.equals(((Address) other).state)); // state check
```
Comment (nishantbudhdev) [inline src/seedu/addressbook/data/person/Address.java]: `state` is never assigned any value. It is not possible to compare as a result, and therefore is semantically incorrect.

**Comment by nishantbudhdev** [comment]
@qhng Great job on finding all the files within the code which require modification.

* You missed out on files in the `\test` folder which need to be modified for regression testing.

Some comments added. Please close the PR after reading comments.

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -25,7 +30,50 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
         }
-        this.value = address;
+        assignDefaultValues();
+        splitAddress(address);
+    }
+    
+    /**
+     * Assigns default values to Block, Street, Unit and Postal Code should they never be initialized.
+     */
+    private void assignDefaultValues() {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: please think through if this is necessary? Java anyway provides with a default constructor that assigns null values to the members.

File: src/seedu/addressbook/data/person/Block.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package seedu.addressbook.data.person;
+
+public class Block {
+
+    public final String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: is it good to make it public? why or why not?

**Pull request closed without merging**